jupyterhub/handlers/base.py

```diff
@@ -366,6 +366,18 @@ def spawner_class(self):
     def spawn_single_user(self, user, server_name='', options=None):
         if server_name in user.spawners and user.spawners[server_name]._spawn_pending:
             raise RuntimeError("Spawn already pending for: %s" % user.name)
+
+        if self.settings['concurrent_spawn_limit'] is not None and \
+           self.hub.pending_spawns > self.settings['concurrent_spawn_limit']:
+            self.log.info(
+                'More than %s pending spawns, throttling',
+                self.settings['concurrent_spawn_limit']
+            )
+            raise web.HTTPError(
+                429,
+                "Too many users are starting now, try again shortly")
```

Comment:
Questions on error message wording: Does the user/admin have to do anything to try again? Will an action need to be taken on the error message?